plugins/cros-ec/fu-cros-ec-usb-device.c

```diff
@@ -316,9 +573,35 @@ fu_cros_ec_usb_device_close (FuUsbDevice *device, GError **error)
 	return TRUE;
 }
 
+static FuFirmware *
+fu_cros_ec_usb_device_prepare_firmware (FuDevice *device,
+					GBytes *fw,
+					FwupdInstallFlags flags,
+					GError **error)
+{
+	FuCrosEcUsbDevice *self = FU_CROS_EC_USB_DEVICE (device);
+	FuCrosEcFirmware *cros_ec_firmware = NULL;
+	g_autoptr(FuFirmware) firmware = fu_cros_ec_firmware_new ();
+
+	fu_device_set_status (device, FWUPD_STATUS_DECOMPRESSING);
+	if (!fu_firmware_parse (firmware, fw, flags, error))
+		return NULL;
+	cros_ec_firmware = FU_CROS_EC_FIRMWARE (firmware);
+
+	/* pick sections */
+	if (!fu_cros_ec_firmware_pick_sections (cros_ec_firmware,
+						self->writeable_offset,
+						error)) {
+		g_prefix_error (error, "failed to pick sections: ");
+		return NULL;
+	}
+	return g_steal_pointer (&firmware);
+}
+
 static void
 fu_cros_ec_usb_device_init (FuCrosEcUsbDevice *device)
 {
+	fu_device_set_protocol (FU_DEVICE (device), "com.google.usb.crosec");
```

Comment:
You should add this to `README.md` too in this PR.

Reply:
And we'll need to add to the LVFS at some point. On that, is `com.google.usb.crosec` *signed* i.e. with some strong public/private method?

Reply:
Added to the README.md.

Hi @hughsie:
I'll tackle getting two versions of servo_micro firmware added to LVFS next.

I have to check with someone who knows more about the signing question. I suspect that some of our end products are signed, but possibly not all.

Reply:
> I suspect that some of our end products are signed, but possibly not all.

Can you tell from asking the hardware? For logitech we actually have two defined protocols, `com.logitech.unifying` and `com.logitech.unifyingsigned` -- that allows us to show the nice green tick for firmware updates that are signed and the red cross for ones that are not.

Reply:
And if not from the hardware, I would think from looking at that payload structure you should be able to tell?

Reply:
Hi @hughsie and @superm1 

So the servo micro board is NOT signed, but some commercially available products that use the cros-ec usb update method are signed.

I'm still investigating what steps I have to take to support signed updates.

What do you recommend for this patch series?

Reply:
My thought is to merge this as is right now. On lvfs mark that protocol as unsigned. When it comes time to support a device with signed payloads modify this plugin to resolve the differences and use a new protocol that we mark as signed.

From my perspective this PR looks great now 🙌 (and I suppose is the last one in this series for this device).

Reply:
Okay, I've added `com.google.usb.crosec` as unsigned on the LVFS for now.